src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Program.cs

```diff
@@ -69,11 +70,18 @@ public static async Task MainAsync(string[] args)
                     .WithOutput(Console.OpenStandardOutput())
                     .ConfigureLogging(builder => builder
                         .AddLanguageServer()
-                        .SetMinimumLevel(logLevel))
+                        .SetMinimumLevel(RazorLSPOptions.GetLogLevelForTrace(trace)))
                     .OnInitialized(async (languageServer, request, response) =>
                     {
                         var fileChangeDetectorManager = languageServer.Services.GetRequiredService<RazorFileChangeDetectorManager>();
                         await fileChangeDetectorManager.InitializedAsync(languageServer);
+
+                        _ = Task.Delay(1000).ContinueWith(async task =>
```

Comment:
Hard to say if it's is a good idea without knowing what the bug being worked around is. Generally for this kind of thing I try to aim for "When X happens or Y milliseconds have passed", where X is something to indicate a ready state. Of course it's not always clear what a good X is, or even that an X is possible given certain limitations.

Reply:
Yup the X in this case should be this method, `OnInitialized`. But it doesn't seem to behave as the name suggest, hence the delay.